yarn-project/aztec.js/src/account/interface.ts

```diff
@@ -0,0 +1,35 @@
+import { Fr } from '@aztec/circuits.js';
+import { AuthWitness, CompleteAddress, FunctionCall, TxExecutionRequest } from '@aztec/types';
+
+// docs:start:account-interface
+/** Creates authorisation witnesses. */
+export interface AuthWitnessProvider {
```

Comment:
The name of the file pains me a little since it is unclear that there are multiple interfaces in here etc 😬 I kinda like the `IName` but this is just my preference really.

Reply:
I wanted to avoid `IName` for consistency, since we're not using it across the codebase, even though it reminds me of my good old C# days. As for the name, agree, it's nasty calling something "Interface" when it's not referring to a language `interface`. I asked ChatGPT and it suggested `Handler` as an alternative, but I don't find it very clear either.